marker_uilints/tests/ui/item/print_me_root_module.stderr

```diff
@@ -0,0 +1,108 @@
+warning: printing item
+  |
+  = note: Mod(
+              ModItem {
+                  data: CommonItemData {
+                      id: ItemId(..),
+                      span: SpanId(..),
+                      vis: Visibility {{ /* WIP: See rust-marker/marker#26 */}},
+                      ident: Ident {
+                          name: "print_me_root_module",
+                          span: $DIR/print_me_root_module.rs:1:1 - 1:1,
+                      },
+                  },
+                  items: [
+                      Use(
+                          UseItem {
+                              data: CommonItemData {
+                                  id: ItemId(..),
+                                  span: SpanId(..),
+                                  vis: Visibility {{ /* WIP: See rust-marker/marker#26 */}},
+                                  ident: Ident {
+                                      name: "",
+                                      span: $DIR/print_me_root_module.rs:1:1 - 1:1,
+                                  },
+                              },
+                              use_path: AstPath {
```

Comment:
I'd be glad if lints didn't even need to see the standard prelude. I guess it deserves it's own ugly crutch to skip such code. And.. in general, I think lints don't ever need to see code with the "built-in" spans auto-generated by the compiler. I imagine someone writes a lint that triggers on a standard prelude use item and thus is cursed to never be able to ignore that false positive unless ignoring it at crate-level meaning the lint is just disabled and *use*less 😺

Reply:
Skipping the import items from the prelude would be pretty simple, items already have an ugly crunch to handle such things.

The question for me, is more if we want this? So these items are definitely not intended for lint emissions, but they are required to get the full information about what namespace etc are available. On the other hand, these changes are always bound to Editions, so if we provide information about that, it should be sufficient to get all the information needed for linting :thinking:

Reply:
The namespace information should be queried from the `TyCtxt`. I don't think that lint authors will specifically traverse all uses and do manually name resolution, which is a complex and recursive algorithm.

Reply:
Done :+1: